               
- Description: xip2: disable readpage and readpages address space operations
  Symptom:     Error in syslog with backchain that warn that readpage or
               readpages address space operation has been called in xip2 file
               system
  Problem:     when calling sys_madvice for a memory area that has been generated
               by calling sys_mmap for a file that is stored on a xip2 file
               system, sys_madvice calls readpage or readpages address space
               operations
  Solution:    disable readpage and readpages which should not be used anyway
  Problem-ID:  
  Archs:       s390-31, s390-64

--- linux-2.5/fs/xip2fs/file.c	4 May 2004 08:29:51 -0000	1.18.2.2
+++ linux-2.5/fs/xip2fs/file.c	4 Jun 2004 10:28:39 -0000	1.18.2.3
@@ -209,8 +209,6 @@
 {
 	struct address_space *mapping = file->f_mapping;
 
-	if (!mapping->a_ops->readpage)
-		return -ENOEXEC;
 	file_accessed(file);
 	vma->vm_ops = &xip2_file_vm_ops;
 	return 0;

--- linux-2.5/fs/xip2fs/inode.c	4 May 2004 08:29:51 -0000	1.27.2.2
+++ linux-2.5/fs/xip2fs/inode.c	4 Jun 2004 10:28:39 -0000	1.27.2.3
@@ -242,6 +242,7 @@
 	return -EROFS;
 }
 
+#if 0
 static int xip2_readpage(struct file *file, struct page *page)
 {
 	struct inode *inode = page->mapping->host;
@@ -304,6 +305,7 @@
 	}
 	return 0;
 }
+#endif
 
 static sector_t xip2_bmap(struct address_space *mapping, sector_t block)
 {
@@ -320,14 +322,14 @@
 }
 
 struct address_space_operations xip2_aops = {
-	.readpage	= xip2_readpage,
-	.readpages	= xip2_readpages,
+//	.readpage	= xip2_readpage,
+//	.readpages	= xip2_readpages,
 	.bmap		= xip2_bmap,
 };
 
 struct address_space_operations xip2_nobh_aops = {
-	.readpage	= xip2_readpage,
-	.readpages	= xip2_readpages,
+//	.readpage	= xip2_readpage,
+//	.readpages	= xip2_readpages,
 	.bmap		= xip2_bmap,
 };
 

